### PR TITLE
Cleanup for IOS-XR driver

### PR DESCRIPTION
**Pull request by mirceaulinic**
<!-- Make sure you have read http://napalm.readthedocs.io/en/latest/contributing/index.html --!>

Too many changes to be displayed...

### CODE REVIEW

**Comment by mirceaulinic** [comment]
@fooelisa any idea why the tests are not executed anymore after adding coveralls https://github.com/napalm-automation/napalm-iosxr/pull/49?

**Comment by mirceaulinic** [comment]
Disregard, just seems to be very slow today :(

**Comment by mirceaulinic** [comment]
Solves https://github.com/napalm-automation/napalm-iosxr/issues/40, https://github.com/napalm-automation/napalm-iosxr/issues/32, https://github.com/napalm-automation/napalm-iosxr/issues/29, https://github.com/napalm-automation/napalm-iosxr/issues/28, https://github.com/napalm-automation/napalm-iosxr/issues/18 and https://github.com/napalm-automation/napalm-iosxr/issues/17

@napalm-automation/council please review :)

File: napalm_iosxr/__init__.py [hunk 1]
```diff
@@ -13,4 +14,14 @@
 # the License.
 
 """napalm_iosxr package."""
-from iosxr import IOSXRDriver
+
+# Import stdlib
+import pkg_resources
+
+# Import local modules
+from iosxr import IOSXRDriver  # noqa
```
Comment (ktbyers) [inline napalm_iosxr/__init__.py]: Should change to:

```
from napalm_iosxr.iosxr import IOSXRDriver
```

The original import is not valid in Python3. I also would drop the # noqa comment and do.

```
__all__ = ['IOSXRDriver']
```

**Comment by ktbyers** [comment]
@mirceaulinic I would change the one **init**.py especially the relative import, but everything else is fine.

File: napalm_iosxr/__init__.py (hunk 1 above)
Comment (mirceaulinic) [reply napalm_iosxr/__init__.py]: Thanks for your suggestions! I've updated as per your comments and I made some more cleanup especially under `get_interfaces_ip`

**Comment by ktbyers** [comment]
@mirceaulinic Why did the one test fail?

**Comment by mirceaulinic** [comment]
> @mirceaulinic Why did the one test fail?

I'm looking into it now @ktbyers - although it's very odd

**Comment by mirceaulinic** [comment]
The classical "it works on my machine":

``` bash
$ nosetests --with-coverage --cover-package napalm_iosxr -v TestIOSXRDriver:TestGetterIOSXRDriver.test_get_bgp_config
test_get_bgp_config (TestIOSXRDriver.TestGetterIOSXRDriver) ... ok

Name                    Stmts   Miss  Cover
-------------------------------------------
napalm_iosxr.py             8      2    75%
napalm_iosxr/iosxr.py     780    599    23%
-------------------------------------------
TOTAL                     788    601    24%
----------------------------------------------------------------------
Ran 1 test in 0.004s

OK
```

File: napalm_iosxr/__init__.py (hunk 1 above)
Comment (mirceaulinic) [reply napalm_iosxr/__init__.py]: All green now @ktbyers!

**Comment by ktbyers** [comment]
+1

**Pull request merged**